client/app/dbaas/logs/detail/roles/edit-permissions/edit-permissions.controller.js

```diff
@@ -0,0 +1,129 @@
+class LogsRolesPermissionsCtrl {
+    constructor ($q, $stateParams, ControllerHelper, CloudMessage, LogsRolesService) {
+        this.$q = $q;
+        this.$stateParams = $stateParams;
+        this.serviceName = this.$stateParams.serviceName;
+        this.roleId = this.$stateParams.roleId;
+        this.ControllerHelper = ControllerHelper;
+        this.LogsRolesService = LogsRolesService;
+        this.CloudMessage = CloudMessage;
+        this.initLoaders();
+    }
+
+    initLoaders () {
```

Comment:
Can you divide what is in this function in multiple functions? It will improve readability, because I'm having a hard time understanding what is going on.

Reply:
This should work now. Please check now :)

Reply:
Yep! Much better now, thank you